### ISSUE TITLE
Google (maybe others?) webfont and "slideDown" issue

### ISSUE THREAD

**Issue by dwhates** [julianshapiro/velocity#636]
It took me a lot of work to track down this bug, so I though I'd share the problem and the solution.

This may be a bit of an edge case, but it happened on three different site designs of mine.

If you have a hidden (display: none) element which you want to then call the "slideDown" function on, if you are using webfonts, and if the webfont you are using has not beet used on any other elements on the page yet, there is a problem where javascript may calculate the height correctly before opening up the element.

Why? I worked out that Chrome (maybe other browsers) doesn't load in that webfont if it's not displayed on the page. Makes sense. So when you call the "slideDown" function, I guess that in the split second it calculated the height of the element it with a fallback font.

I fixed this by forcing the webfont to load up by placing it in a ghost element (position: fixed, opacity: 0) element on the page. That forces Chrome to load the webfont on first page load. Then when we use the "slideDown", everything is calculated as expected.

Maybe a bit much to load up the webfont if there is a chance the element will never be "slideDowned". But I don't know how to otherwise do it and I'm just glad I finally got it to work!

**Comment by Rycochet** [comment]
Technically this isn't a bug, but still worth talking/thinking about ;-)

There's no way to force loading a webfont without actually creating a visible (even if `visibility:hidden`) element using it, and due to the way that webfonts often use unicode codepage character sets and only load the specific files that are needed I think this might be outside the scope of Velocity directly - saying that, mentioning it in the docs at least would be useful.

As a workaround make use of `visibility:hidden` rather than `display:none`.

Something that might be within Velocity could be a way to force a running animation to recalculate sizes - though as it's only really dealing with pixel sizes rather than relative sizes that might be a bit hard to do without changing quite a lot of code.

**Comment by dwhates** [comment]
I suppose it's only a bug in that, the intended behavior of smooth scrolling doesn't work. It expands and then, when at the end of the transition Velocity  updates the element to {height: auto}, it jerks to that as it is different from the height it has transitioned to.

Is there a way in javascript to check if a certain resource is loading, without that causing too much strain on anything?

**Comment by dwhates** [comment]
```javascript
if( webfontLoadCheck() ) {
    el.velocity("slideDown")
}
```

**Comment by Rycochet** [comment]
There's not actually any way of telling if a font has been loaded, however there are a lot of workarounds to do so - generally it's a case of creating an element with text in it, and watching for that text size changing - the downside being unknown time and unknown failure timeouts etc - stackoverflow is your friend there at least ;-)

Possibly adding `will-change: display;` to an element with the needed font it might load it in advance (see http://caniuse.com/#feat=will-change ) - however I don't know if that would force the font to be loaded even assuming that it's supported.

**Comment by dwhates** [comment]
Yeah, creating an element and loading it in is the way I'm doing it at the moment. It's a workaround I can live with for now. As you said, probably nothing worth sweating over, other than to mention it in the docs to avoid placing other developers in the same situation I was in; i.e. blindly hunting around for the root of the problem for hours on end! :)

**Comment by Rycochet** [comment]
Needs adding to documentation, though simply a browser issue and not something we can workaround.
